Deadlock.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CodeReviewDemo
+{
+    internal class DeadlockDemo
+    {
+        object myLock1 = new object();
```

Comment:
Consider renaming `myLock1` and `myLock2` to more descriptive names, such as `resourceLock1` and `resourceLock2`, to improve code readability.